contracts/src/v0.8/shared/token/ERC677/BurnMintERC677.sol

```diff
@@ -1,217 +1,33 @@
-  // |                      Burning & minting                       |
-  // ================================================================
-
-  /// @inheritdoc ERC20Burnable
-  /// @dev Uses OZ ERC20 _burn to disallow burning from address(0).
-  /// @dev Decreases the total supply.
-  function burn(uint256 amount) public override(IBurnMintERC20, ERC20Burnable) onlyBurner {
-    super.burn(amount);
-  }
-
-  /// @inheritdoc IBurnMintERC20
-  /// @dev Alias for BurnFrom for compatibility with the older naming convention.
-  /// @dev Uses burnFrom for all validation & logic.
-  function burn(address account, uint256 amount) public virtual override {
-    burnFrom(account, amount);
-  }
-
-  /// @inheritdoc ERC20Burnable
-  /// @dev Uses OZ ERC20 _burn to disallow burning from address(0).
-  /// @dev Decreases the total supply.
-  function burnFrom(address account, uint256 amount) public override(IBurnMintERC20, ERC20Burnable) onlyBurner {
-    super.burnFrom(account, amount);
-  }
-
-  /// @inheritdoc IBurnMintERC20
-  /// @dev Uses OZ ERC20 _mint to disallow minting to address(0).
-  /// @dev Disallows minting to address(this)
-  /// @dev Increases the total supply.
-  function mint(address account, uint256 amount) external override onlyMinter validAddress(account) {
-    if (i_maxSupply != 0 && totalSupply() + amount > i_maxSupply) revert MaxSupplyExceeded(totalSupply() + amount);
-
-    _mint(account, amount);
-  }
-
-  // ================================================================
-  // |                            Roles                             |
-  // ================================================================
-
-  /// @notice grants both mint and burn roles to `burnAndMinter`.
-  /// @dev calls public functions so this function does not require
-  /// access controls. This is handled in the inner functions.
-  function grantMintAndBurnRoles(address burnAndMinter) external {
-    grantMintRole(burnAndMinter);
-    grantBurnRole(burnAndMinter);
-  }
-
-  /// @notice Grants mint role to the given address.
-  /// @dev only the owner can call this function.
-  function grantMintRole(address minter) public onlyOwner {
-    if (s_minters.add(minter)) {
-      emit MintAccessGranted(minter);
-    }
-  }
-
-  /// @notice Grants burn role to the given address.
-  /// @dev only the owner can call this function.
-  function grantBurnRole(address burner) public onlyOwner {
-    if (s_burners.add(burner)) {
-      emit BurnAccessGranted(burner);
-    }
-  }
-
-  /// @notice Revokes mint role for the given address.
-  /// @dev only the owner can call this function.
-  function revokeMintRole(address minter) public onlyOwner {
-    if (s_minters.remove(minter)) {
-      emit MintAccessRevoked(minter);
+contract BurnMintERC677 is BurnMintERC20, IERC677 {
+  constructor(
+    string memory name,
+    string memory symbol,
+    uint8 decimals_,
+    uint256 maxSupply_
+  ) BurnMintERC20(name, symbol, decimals_, maxSupply_) {}
```

Comment:
nit: I think we can drop the `_`s here. In general, I think apply the campsite rule here "leave it better than you found it". This is old code that we're touching for the first time in a while, so it's nice to use the opportunity to bring the code quality up to par with our style guide :)

Reply:
I thought the campsite rule was "leave no trace" in which case changing nothing would be key but sure I can change it no problem